### ISSUE TITLE
Mutability of TF_NewTensor’s arguments

### ISSUE THREAD

**Issue by IvanUkhov** [tensorflow/tensorflow#3176]
The C API provides the [following function](https://github.com/tensorflow/tensorflow/blob/v0.9.0/tensorflow/core/public/tensor_c_api.h#L195) for creating tensors:

```c
extern TF_Tensor* TF_NewTensor(TF_DataType, long long* dims, int num_dims,
                               void* data, size_t len,
                               void (*deallocator)(void* data, size_t len,
                                                   void* arg),
                               void* deallocator_arg);
```

Based on what the function does conceptually and on how it is [implemented](https://github.com/tensorflow/tensorflow/blob/v0.9.0/tensorflow/core/client/tensor_c_api.cc#L122), it doesn’t seem to be reasonable that `dims` is required to be a pointer to mutable data. This makes the usage of the function a bit inconvenient, especially when developing bindings for other languages. If it’s OK to change the signature of the function, I can submit a pull request. Thanks!

Regards,
Ivan

**Comment by josh11b** [comment]
That sounds like a fine change to me.

**Comment by IvanUkhov** [comment]
Fixed in #3214.
